command/agent/check.go

```diff
@@ -532,6 +545,33 @@ func (c *CheckTCP) check() {
 	c.Notify.UpdateCheck(c.CheckID, structs.HealthPassing, fmt.Sprintf("TCP connect %s: Success", c.TCP))
 }
 
+// checkUDP is invoked periodically to perform the UDP check
+func (c *CheckTCP) checkUDP() {
+	conn, err := c.dialer.Dial(`udp`, c.UDP)
+	if err == nil {
+		b := []byte("CONNECTED-MODE SOCKET")
```

Comment:
The payload here I pulled from the Golang UDP test suite.  The other monitoring examples I linked above use varying payloads, some send multiple packets for redundancy in case the first was dropped in transit.  It would be nice to customize the payload but IIRC none of the other Consul health checks offer customizable payloads?  At that point I may have to further separate this UDP check from the TCP check rather than piggybacking on the TCP logic.